sklearn/tree/tests/test_tree.py

```diff
@@ -260,22 +260,22 @@ def test_iris():
             "".format(name, criterion, score))
 
 
-def test_boston():
-    # Check consistency on dataset boston house prices.
+def test_diabetes():
+    # Check consistency on diabetes dataset.
 
     for (name, Tree), criterion in product(REG_TREES.items(), REG_CRITERIONS):
         reg = Tree(criterion=criterion, random_state=0)
-        reg.fit(boston.data, boston.target)
-        score = mean_squared_error(boston.target, reg.predict(boston.data))
+        reg.fit(diabetes.data, diabetes.target)
+        score = mean_squared_error(diabetes.target, reg.predict(diabetes.data))
         assert score < 1, (
             "Failed with {0}, criterion = {1} and score = {2}"
             "".format(name, criterion, score))
 
         # using fewer features reduces the learning ability of this tree,
         # but reduces training time.
         reg = Tree(criterion=criterion, max_features=6, random_state=0)
-        reg.fit(boston.data, boston.target)
-        score = mean_squared_error(boston.target, reg.predict(boston.data))
+        reg.fit(diabetes.data, diabetes.target)
+        score = mean_squared_error(diabetes.target, reg.predict(diabetes.data))
```

Comment:
Here, we might already be in the regime `score > 0` because we are using a subset of features at each split.
We can observe something really similar if you limit the depth of the tree.

Reply:
The score here is always 0 for both Boston and diabetes dataset

Reply:
ok could you limit the depth as well then `max_depth=8` and check if the MSE is increasing. We can reword the comment above 

```
        # using fewer features and limiting the tree depth reduces the
        # learning ability of this tree, but reduces training time.
```